samples/msgext-ai-sentiment-analysis/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler } = require("botbuilder");
+const { Configuration, OpenAIApi } = require("openai");
+var textToAnalyze = "";
+var sentimentResponse = "";
+
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+        this.baseUrl = process.env.BaseUrl;
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+
+            for (let member = 0; member < membersAdded.length; member++) {
+                if (membersAdded[member].id !== context.activity.recipient.id) {
+                    await context.sendActivity("Hello and welcome. This sample helps you to send sentiment for any text message");
+                }
+            }
+
+            await next();
+        });
+    }
+
+    // fetch and handle message extension task module 
+    async handleTeamsMessagingExtensionFetchTask(context, action) {
+
+        textToAnalyze = action.messagePayload.body.content;
+        textToAnalyze = textToAnalyze.replace(/(<([^>]+)>)/ig, '');
+
+        // Configure your apikey
+        const configuration = new Configuration({
+            apiKey: process.env.apiKey
+        });
+
+        // Updaitng configuration with OpenAI API
+        const openai = new OpenAIApi(configuration);
+
+        const analyzeSentiment = async (tweet) => {
+            try {
+                const messages = [
+                    {
+                        role: 'system',
+                        content: `You will be provided with a tweet, and your task is to classify its sentiment as positive, neutral, or negative`,
+                    },
+                    {
+                        role: 'user',
+                        content: `${tweet}`
+                    }
+                ]
+
+                // CreateChatCompletion api call
+                const response = await openai.createChatCompletion({
+                    model: "gpt-3.5-turbo",
```

Comment:
We need to use Deployed model, not sure how it is working.
Refer my Enterprise search app.